src/stim/diagram/detector_slice/detector_slice_set.cc

```diff
@@ -682,9 +713,23 @@ void DetectorSliceSet::write_svg_contents_to(
             bool drawCorners = false;
             if (color == nullptr) {
                 drawCorners = true;
-                color = "#AAAAAA";
+                color = BG_GREY;
             }
 
+            // Open the group element for this slice
+            out << "<g id=\"slice:" << target.val();
+            bool first = true;
+            for (auto c : detector_coordinates.at(target.val())){
```

Comment:
Use `const auto &` instead of `auto` in loops, to avoid accidental copies and mutations. I think this loop is currently unnecessarily copy constructing every element iterated over.